lib/src/components/NxModal/__tests__/NxModal.test.tsx

```diff
@@ -33,17 +38,50 @@ describe('NxModal', function() {
 
   it('merges any passed in className to the nx-modal div', function() {
     const nxModal = shallow(
-      <NxModal className="test"/>
+      <NxModal className="test" onClose={dummyCloseHandler}/>
     );
     const nxModalDiv = nxModal.find('.nx-modal');
     expect(nxModalDiv.hasClass('test')).toEqual(true);
   });
 
   it('includes any passed in attributes to the nx-modal div', function() {
     const nxModal = shallow(
-      <NxModal id="modal-id" data-foo="bar"/>
+      <NxModal id="modal-id" data-foo="bar" onClose={dummyCloseHandler}/>
     );
     expect(nxModal.find('.nx-modal').prop('id')).toEqual('modal-id');
     expect(nxModal.find('.nx-modal').prop('data-foo')).toEqual('bar');
   });
+
+  describe('NxModal event listener support', () => {
+    let container: HTMLDivElement | null;
+
+    it('executes onClose method when pressing ESC key', function () {
+      const mockCallBack = jest.fn();
+      const nxModal = (
+        <div>
```

Comment:
Is this div necessary

Reply:
> Is this div necessary

Div was not necessary. Refactored and removed creator function: 3f1b2a4